### PR TITLE
Add TypeScript declarations to iron-meta.

### PR DESCRIPTION
**Pull request by aomarks**
This PR adds TypeScript declarations generated by https://github.com/Polymer/gen-typescript-declarations/

These declarations can be re-generated by running `npm run update-types`.

Tracker: https://github.com/Polymer/gen-typescript-declarations/issues/79

### CODE REVIEW

File: iron-meta.d.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ *     document.createElement('iron-meta').byKey('info');
+ *
+ * Or, in a Polymer element, you can include a meta in your template:
+ *
+ *     <iron-meta id="meta"></iron-meta>
+ *     ...
+ *     this.$.meta.byKey('info');
+ */
+interface IronMetaElement extends Polymer.Element {
```
Comment (e111077) [inline iron-meta.d.ts]: Polymer.IronMeta is not being typed, only the element.

https://github.com/PolymerElements/iron-meta/blob/master/iron-meta.html#L66

File: iron-meta.d.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * The meta-data to store or retrieve.
+   */
+  value: any;
+
+  /**
+   * If true, `value` is set to the iron-meta instance itself.
+   */
+  self: boolean|null|undefined;
```
Comment (e111077) [inline iron-meta.d.ts]: is `__meta` being ignored because the name suggests it's private or because it's computed?

Comment (aomarks) [reply iron-meta.d.ts]: Yep, Analyzer assumes that double leading underscore means private.

https://github.com/Polymer/polymer-analyzer/blob/209980d52467915d73331f775c8bc8800243e6f4/src/javascript/esutil.ts#L403

File: iron-meta.d.ts (hunk 1 above)
Comment (aomarks) [reply iron-meta.d.ts]: Analyzer doesn't pick up classes defined this way, so added manual typings for it, PTAL!

**Pull request merged**